test/source-map-uploader-plugin.test.js

```diff
@@ -166,10 +166,55 @@ if (process.env.WEBPACK_VERSION !== '3') {
     }, err => {
       if (err) end(err)
     })
+
+    test('it ignores source maps for css files by default', t => {
+      t.plan(7)
+      t.plan(3)
+      const requests = []
+      const end = err => {
+        clearTimeout(timeout)
+        server.close()
+        if (err) return t.fail(err.message)
+        t.end()
+      }
+
+      // prevent test hanging forever
+      const timeout = setTimeout(end, 10000)
+
+      const done = () => {
+        t.equal(requests[0].minifiedUrl, '*/dist/main.js')
+        t.equal(requests[0].parts[0].filename, 'main.js.map')
+        t.equal(requests[0].parts[1].filename, 'main.js')
+        end()
+      }
+
+      const server = http.createServer((req, res) => {
+        parseFormdata(req, function (err, data) {
+          if (err) {
+            res.end('ERR')
+            return end(err)
+          }
+          requests.push({
+            apiKey: data.fields.apiKey,
+            minifiedUrl: data.fields.minifiedUrl,
+            parts: data.parts.map(p => ({ name: p.name, filename: p.filename }))
+          })
+          res.end('OK')
+          done()
+        })
+      })
+      server.listen()
+      exec(`${__dirname}/../node_modules/.bin/webpack`, {
+        env: Object.assign({}, process.env, { PORT: server.address().port }),
+        cwd: `${__dirname}/fixtures/e`
+      }, (err) => {
+        if (err) end(err)
+      })
+    })
   })
 
-  test('it works when plugins cause a chunk to have multiple source maps', t => {
-    t.plan(7)
+  test('it uploads css map files if you really want', t => {
```

Comment:
In addition to removing items from the blacklist, it'd be worth testing adding items to the blacklist, such as `.php` files like in this test.

Reply:
the presence of items in the ignore list is tested with the "it ignores source maps for css files by default" case

Reply:
I see that as just testing the default though, not whether new items can be added.  I realise the distinction is minimal, but think testing three different scenarios (default, added to, removed from) would be a good idea.

Reply:
It seems unreasonably hard to make webpack output a js file with a non-JS extension _AND_ create a source map. I can get it to have a different extension (e.g. `.bundle`) but then it doesn't output a source map.

The way I see it – the option is not added to or removed from, it's simply replaced. So the tests still give assurance that it iterates the array and sends/does not send based on the presence of extensions within it.

You ok if I don't add a test for this specific case?

Reply:
Yeah, that's fair.  I assumed with the addition of `.php` it was a common occurrence.